### PR TITLE
Signing Blinded/BlobSidecar requests

### LINKED ISSUE #883: Implement Blinded/Blobsidecar signing in Web3Signer
Implement Blinded/Blobsidecar signing in Web3Signer.

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Implement signing of BlindedBlobSidecar and BlobSidecar requests. 
## Fixed Issue(s)

Fixes #883 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

## Testing

- [x] I thought about testing these changes in a realistic/non-local environment.

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2RequestUtils.java [hunk 1]
```diff
@@ -533,4 +426,36 @@ private static SafeFuture<Bytes> signingRootFromSyncCommitteeUtils(
...
+          new DataStructureUtil(spec).randomBlindedBlobSidecar();
+      signingRoot =
+          new SigningRootUtil(spec)
+              .signingRootForBlindedBlobSidecar(tekuBlindedBlobSidecar, tekuForkInfo);
+      blindedBlobSidecar = fromInternalBlindedBlobSidecar(tekuBlindedBlobSidecar);
+    } else {
+      final BlobSidecar tekuBlobSidecar = new DataStructureUtil(spec).randomBlobSidecar();
+      signingRoot =
+          new SigningRootUtil(spec).signingRootForBlobSidecar(tekuBlobSidecar, tekuForkInfo);
+      blindedBlobSidecar = fromInternalBlobSidecar(tekuBlobSidecar);
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2RequestUtils.java:452]: This looks like it building a blinded blob side car in both cases. Is there a difference between the two?

**Review by gfukushima — APPROVED** [review]
lgtm, few spdx to fix only

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/KZGProof.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ */
+package tech.pegasys.web3signer.core.service.http.handlers.signing.eth2;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Objects;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public class KZGProof {
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/KZGProof.java:21]: we don't have Teku release with `(api.schema) KZFProof` yet.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2SigningRequestBodyBuilder.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    return this;
+  }
+
+  public Eth2SigningRequestBodyBuilder withValidatorRegistration(
+      ValidatorRegistration validatorRegistration) {
+    this.validatorRegistration = validatorRegistration;
+    return this;
+  }
+
+  public Eth2SigningRequestBodyBuilder withBlindedBlobSidecar(
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2SigningRequestBodyBuilder.java:137]: To be consistent with the API this should withBlobSidecar not withBlindedBlobSidecar

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SigningRequestBody.java [hunk 4]
```diff
@@ -20,135 +20,24 @@
...
+    @JsonProperty("aggregate_and_proof") AggregateAndProof aggregateAndProof,
+    @JsonProperty("voluntary_exit") VoluntaryExit voluntaryExit,
+    @JsonProperty("randao_reveal") RandaoReveal randaoReveal,
+    @JsonProperty("deposit") DepositMessage deposit,
+    @JsonProperty("sync_committee_message") SyncCommitteeMessage syncCommitteeMessage,
+    @JsonProperty("sync_aggregator_selection_data")
+        SyncAggregatorSelectionData syncAggregatorSelectionData,
+    @JsonProperty("contribution_and_proof") ContributionAndProof contributionAndProof,
+    @JsonProperty("validator_registration") ValidatorRegistration validatorRegistration,
+    @JsonProperty("blob_sidecar") BlindedBlobSidecar blindedBlobSidecar) {}
```
Comment (gfukushima) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SigningRequestBody.java:43]: blob_sidecar or blindedBlobSidecar?

Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SigningRequestBody.java:43]: `@JsonProperty("blob_sidecar")` is intended. However, I'll be changing `BlindedBlobSidecar` to `BlobSidecar`.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2SigningRequestBodyBuilder.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.Eth2SigningRequestBody;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.ForkInfo;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.RandaoReveal;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.SyncAggregatorSelectionData;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.SyncCommitteeMessage;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.ValidatorRegistration;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public final class Eth2SigningRequestBodyBuilder {
```
Comment (siladu) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2SigningRequestBodyBuilder.java:34]: 👍

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/BlsBlindedBlobcarSigningAcceptanceTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class BlsBlindedBlobcarSigningAcceptanceTest extends SigningAcceptanceTestBase {
```
Comment (siladu) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/BlsBlindedBlobcarSigningAcceptanceTest.java:40]: Think we should also remove Blinded from this name

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/BlsBlindedBlobcarSigningAcceptanceTest.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+  private static final String PRIVATE_KEY =
+      "3ee2224386c82ffea477e2adf28a2929f5c349165a4196158c7f3a2ecca40f35";
+  private static final MetadataFileHelpers METADATA_FILE_HELPERS = new MetadataFileHelpers();
+  private static final BLSSecretKey KEY =
+      BLSSecretKey.fromBytes(Bytes32.fromHexString(PRIVATE_KEY));
+  private static final BLSKeyPair KEY_PAIR = new BLSKeyPair(KEY);
+  private static final BLSPublicKey PUBLIC_KEY = KEY_PAIR.getPublicKey();
+
+  @ParameterizedTest(name = "#{index} - Blinded Blobsidecar Signing: {0}")
```
Comment (siladu) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/BlsBlindedBlobcarSigningAcceptanceTest.java:50]: and here
```suggestion
  @ParameterizedTest(name = "#{index} - Blobsidecar Signing: {0}")
```

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlobSidecar.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.tuweni.bytes.Bytes32;
+
+/**
+ * Json representation of BlindedBlobSidecar and BlobSidecar. The blob_root is the "hash tree root"
+ * of BlobSidecar's blob
+ *
+ * @param blockRoot Block root
```
[the remark this answers is not in the document]
Comment (siladu) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlobSidecar.java:28]: Is @JsonProperty or the record causing these alerts somehow?

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlobSidecar.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    @JsonProperty("slot") UInt64 slot,
+    @JsonProperty("block_parent_root") Bytes32 blockParentRoot,
+    @JsonProperty("proposer_index") UInt64 proposerIndex,
+    @JsonProperty("blob_root") Bytes32 blobRoot,
+    @JsonProperty("kzg_commitment") KZGCommitment kzgCommitment,
+    @JsonProperty("kzg_proof") KZGProof kzgProof) {
+
+  public tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlindedBlobSidecar
+      asInternalBlindedBlobSidecar(final SpecVersion spec) {
+    final BlindedBlobSidecarSchema blindedBlobSidecarSchema =
```
Comment (siladu) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlobSidecar.java:49]: Should this be BlobSidecar type instead now?

**Review by siladu — COMMENTED** [review]
Looks pretty straight forward, mostly just questions/renames. 

It's a shame w3s needs to know about blinded blocks...is that just because of the Teku class hierarchy dependency?

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlobSidecar.java (hunk 9 above)
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlobSidecar.java:49]: this is internal (Teku) class.

**Comment by usmansaleem** [comment]
> Looks pretty straight forward, mostly just questions/renames.
> 
> It's a shame w3s needs to know about blinded blocks...is that just because of the Teku class hierarchy dependency?

Correct, Teku's classes are used to calculate signing root, hence we need them.

**Review by siladu — APPROVED** [review]

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2RequestUtils.java [hunk 10]
```diff
@@ -533,4 +426,37 @@ private static SafeFuture<Bytes> signingRootFromSyncCommitteeUtils(
...
+          new SigningRootUtil(spec)
+              .signingRootForBlindedBlobSidecar(tekuBlindedBlobSidecar, tekuForkInfo);
+      blobSidecar = fromInternalBlindedBlobSidecar(tekuBlindedBlobSidecar);
+    } else {
+      // generate random blobsidecar
+      final BlobSidecar tekuBlobSidecar = new DataStructureUtil(spec).randomBlobSidecar();
+      signingRoot =
+          new SigningRootUtil(spec).signingRootForBlobSidecar(tekuBlobSidecar, tekuForkInfo);
+      blobSidecar = fromInternalBlobSidecar(tekuBlobSidecar);
+    }
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2RequestUtils.java:454]: Still don't understand why we are testing two cases. They are the same as far as Web3Signer is concerned. Are we gaining anything but having the other test case?

File: gradle/versions.gradle [hunk 11]
```diff
@@ -134,7 +134,7 @@ dependencyManagement {
     dependency 'com.azure:azure-core-http-netty:1.13.5'
 
     dependency 'com.zaxxer:HikariCP:5.0.1'
-    dependency 'org.postgresql:postgresql:42.5.3'
+    dependency 'org.postgresql:postgresql:42.6.0'
```
Comment (jframe) [inline gradle/versions.gradle:137]: Was this fix an issue? Probably worthy of a changelog entry if that's the case

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2RequestUtils.java (hunk 10 above)
Comment (usmansaleem) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/Eth2RequestUtils.java:454]: You are right, I was still thinking with regards to Teku side of things ... I'll fix the test case to test only "BlobSidecar"

File: gradle/versions.gradle (hunk 11 above)
Comment (usmansaleem) [reply gradle/versions.gradle:137]: This was originally to fix an issue reported by owasp dependency-check plugin, however, it was later turned out to be false positive i.e. PostgreSQL 12 is affected, not the JDBC driver. I am going to revert this back to `42.5.3`. The JDBC driver should be update from a separate PR.

**Review by jframe — APPROVED** [review]

**Pull request merged**